firstvoices/backend/views/word_of_the_day_views.py

```diff
@@ -140,10 +146,19 @@ def get_selected_word(self, site_slug):
     def list(self, request, *args, **kwargs):
         # Overriding list method from FVPermissionViewSetMixin to only get the first word
         site = self.get_validated_site()
-
-        # Logic to select queryset
         selected_word = self.get_selected_word(site[0].slug)
-        queryset = utils.filter_by_viewable(request.user, selected_word)
+
+        queryset = WordOfTheDay.objects.none()
+
+        if selected_word:
+            unfiltered_queryset = DictionaryEntry.objects.filter(
+                id=selected_word[0].dictionary_entry.id
+            )
+            filtered_queryset = utils.filter_by_viewable(
+                request.user, unfiltered_queryset
+            )
+            if filtered_queryset:
+                queryset = selected_word
```

Comment:
Won't this just return no word of the day if the user doesn't have the ability to see the selected word of the day and is that the functionality we want? I'm just wondering if a different word of the day should be selected if a user doesn't have permission to see the current word.

Reply:
Yes, that is by design. A few cases can arise where the visibility of the updated word does not allow it to be viewed by the user, but since the word of the day can only be chosen once per date, we cannot select/show an alternative word and thus no word is returned.